Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift

```diff
@@ -51,7 +51,7 @@ final class NetworkProtectionConnectionTester {
     static let monitorQueue = DispatchQueue(label: "com.duckduckgo.NetworkProtectionConnectionTester.monitorQueue")
     static let endpoint = NWEndpoint.hostPort(host: .name("www.duckduckgo.com", nil), port: .https)
 
-    private var timer: DispatchSourceTimer?
+    private var task: Task<Never, Error>?
```

Comment:
Replaced the tester timer with a task, since my understanding is `DispatchSourceTimer` is more prone to deadlocks than a task is.